CONTRIBUTING.md

```diff
@@ -1,152 +1,196 @@
+- git
 
-### Pull Request Reviews
+Start by getting the source code:
 
-It is the responsibility of the author to ask for at least one person to review their Pull Request. That person should know the area of the code being changed. If the chosen reviewer does not feel confident in the review, they can then ask for someone else to additionally look at the code.
+```bash
+$ git clone --recurse-submodules https://github.com/raiden-network/light-client.git
+$ cd light-client
+```
 
-All the developers in the team should perform Pull Request reviews. Make it a habit to check [this](https://github.com/raiden-network/light-client/pulls?q=is%3Apr+is%3Aopen+label%3A%22dev%3A+Please+Review%22) link often to help your fellow colleagues who have PRs open pending for review.
+### Code style
 
-We have a lot of tools that automatically check the quality of the code (eslint, prettier). All these are automatically ran by the CI. Therefore fixes related to linting are not usually part of PR reviews. Additionally reviewers are encouraged to not be nitpicky about the suggested changes they ask from the PR author. If something is indeed nitpicky then the reviewer is encouraged to state it beforehand. Example:
+The code style is enforced by [ESLint](https://eslint.org), which means that in most cases you don't need to do anything other than running:
 
-> nitpick: I don't really think XYZ makes sense here. If possible it would be nice to have it changed to KLM
+```bash
+$ npm run lint
+```
 
-The author of the PR can then choose to implement the nitpicks or ignore them.
+To automatically fix any fixable codestyle issue in the SDK or Light Client, you may add to the `lint` command the `--fix` option.
 
-PR authors should make pull request reviews easier. Make them as small as possible and even if some code is touched it does not mean that it needs to be refactored. For example don't mix style/typing changes with a big PR.
+#### Observables
 
-When a reviewer starts a PR review he should write a comment in the PR stating he is doing so. For example:
+Functions that return observables should have `$` as suffix, e.g.:
 
-> Reviewing this now
+```javascript
+myFunction$(): Observable<RaidenAction>
+```
 
-This is to keep track of who is reviewing a PR and to also know when a PR review is ongoing.
+#### Vue components
 
-For complicated PRs that touch the core of the protocol at least 2 core developers are recommended to have a look and provide an opinion.
+When using custom vue components in templates make sure to use the kebab case name of the components.
 
-When performing a PR review of non trivial PRs it is recommended to clone the branch locally, explore the changes with your editor, run tests and experiment with the changes so that a better understanding of the code change can be achieved and good constructive feedback given back to the author.
+👎 Don't do this:
 
-## Development environment setup
+```html
```

Comment:
@nephix don't we also enforce this with eslint? If this is part of the eslint configuration maybe we should remove it.

Reply:
I don't know, could be. I just left it inside because I wasn't sure

Reply:
@nephix https://github.com/raiden-network/light-client/pull/599 :P

Reply:
Cool, time to get rid of it :)